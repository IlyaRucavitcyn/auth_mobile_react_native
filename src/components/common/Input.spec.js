import 'react-native';
import { TextInput } from 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { Input } from './Input';

describe("Input element testing", () => {

    let props;
    const label = "Testing label";
    const value = "";
    const placeholder = "Testing placeholder";
    const secureTextEntry = null;
    const editable = null;

    beforeEach(() => {
        props = {
            label,
            value,
            placeholder,
            secureTextEntry,
            editable,
            onChangeText: jest.fn()
        }
    });

    test('renders snapshot correctly', () => {
        const wrapper = shallow(
            <Input {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('passes props correctly', () => {
        const updatedProps = {
            label: "New test label",
            placeholder: "New test placeholder",
            secureTextEntry: true,
            editable: true
        }
        Object.assign(props, updatedProps)
        const wrapper = shallow(
            <Input {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('runs eventListener when input has been changed', () => {
        const wrapper = shallow(
            <Input
                {...props} />
        );
        wrapper.find(TextInput).simulate("ChangeText", "test text");
        expect(props.onChangeText).toHaveBeenCalledWith("test text");
    });

});