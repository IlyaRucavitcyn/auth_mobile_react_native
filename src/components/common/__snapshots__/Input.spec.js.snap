// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input element testing passes props correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    editable={true}
    label="New test label"
    onChangeText={[Function]}
    placeholder="New test placeholder"
    secureTextEntry={true}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        >
          New test label
        </Text>,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          editable={true}
          onChangeText={[Function]}
          placeholder="New test placeholder"
          secureTextEntry={true}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          value=""
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "New test label",
          "ellipsizeMode": "tail",
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": "New test label",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "editable": true,
          "onChangeText": [Function],
          "placeholder": "New test placeholder",
          "secureTextEntry": true,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          >
            New test label
          </Text>,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            onChangeText={[Function]}
            placeholder="New test placeholder"
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            value=""
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "New test label",
            "ellipsizeMode": "tail",
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": "New test label",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "editable": true,
            "onChangeText": [Function],
            "placeholder": "New test placeholder",
            "secureTextEntry": true,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Input element testing renders snapshot correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    editable={null}
    label="Testing label"
    onChangeText={[Function]}
    placeholder="Testing placeholder"
    secureTextEntry={null}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            }
          }
        >
          Testing label
        </Text>,
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          editable={null}
          onChangeText={[Function]}
          placeholder="Testing placeholder"
          secureTextEntry={null}
          style={
            Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
          value=""
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 40,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Testing label",
          "ellipsizeMode": "tail",
          "style": Object {
            "flex": 1,
            "fontSize": 18,
            "paddingLeft": 20,
          },
        },
        "ref": null,
        "rendered": "Testing label",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCorrect": false,
          "editable": null,
          "onChangeText": [Function],
          "placeholder": "Testing placeholder",
          "secureTextEntry": null,
          "style": Object {
            "color": "#000",
            "flex": 2,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "flex": 1,
                "fontSize": 18,
                "paddingLeft": 20,
              }
            }
          >
            Testing label
          </Text>,
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={null}
            onChangeText={[Function]}
            placeholder="Testing placeholder"
            secureTextEntry={null}
            style={
              Object {
                "color": "#000",
                "flex": 2,
                "fontSize": 18,
                "lineHeight": 23,
                "paddingLeft": 5,
                "paddingRight": 5,
              }
            }
            value=""
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Testing label",
            "ellipsizeMode": "tail",
            "style": Object {
              "flex": 1,
              "fontSize": 18,
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": "Testing label",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCorrect": false,
            "editable": null,
            "onChangeText": [Function],
            "placeholder": "Testing placeholder",
            "secureTextEntry": null,
            "style": Object {
              "color": "#000",
              "flex": 2,
              "fontSize": 18,
              "lineHeight": 23,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
